### PR TITLE
Added MvxActionBarDrawerToggle

### PR DESCRIPTION
**Pull request by martijn00**

### CODE REVIEW

**Comment by martijn00** [comment]
Looks fine to me!

File: Cirrious.MvvmCross.Droid.Support.AppCompat/MvxActionBarDrawerToggle.cs [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+
+        public event ActionBarDrawerChangedEventHandler DrawerOpened;
+
+        public event ActionBarDrawerChangedEventHandler DrawerSlide;
+
+        public event ActionBarDrawerChangedEventHandler DrawerStateChanged;
+
+        public override void OnDrawerClosed(View drawerView)
+        {
+            if (null != this.DrawerClosed)
```
Comment (jamie94bc) [inline Cirrious.MvvmCross.Droid.Support.AppCompat/MvxActionBarDrawerToggle.cs:56]: This isn't thread safe. It should be:

``` c#
var handler = this.DrawerClosed;
if (handler != null) {
    handler(this, new ActionBarDrawerEventArgs { DrawerView = drawerView });
}
```

Or in C#6...

``` c#
this.DrawerClosed?.Invoke(this, new ActionBarDrawerEventArgs { DrawerView = drawerView });
```

File: Cirrious.MvvmCross.Droid.Support.AppCompat/MvxActionBarDrawerToggle.cs [hunk 2]
```diff
@@ -0,0 +1,90 @@
+﻿using System;
+using Android.App;
+using Android.Views;
+using Android.Support.V4.Widget;
+using Android.Support.V7.Widget;
+
+namespace Cirrious.MvvmCross.Droid.AppCompat
+{
+    public class ActionBarDrawerEventArgs : EventArgs
```
Comment (jamie94bc) [inline Cirrious.MvvmCross.Droid.Support.AppCompat/MvxActionBarDrawerToggle.cs:9]: Is it worth splitting this into separate classes? Here's what I currently use:

``` c#
public class ActionBarDrawerEventArgs : EventArgs {
    public View DrawerView { get; set; }

    public ActionBarDrawerEventArgs(View drawerView) {
        this.DrawerView = drawerView;
    }
}

public sealed class ActionBarDrawerSlideEventArgs : ActionBarDrawerEventArgs {
    public float SlideOffset { get; set; }

    public ActionBarDrawerSlideEventArgs(View drawerView, float slideOffset) : base(drawerView) {
        this.SlideOffset = slideOffset;
    }
}

public sealed class ActionBarDrawerStateChangeEventArgs : EventArgs {
    public int NewState { get; set; }

    public ActionBarDrawerStateChangeEventArgs(int newState) {
        this.NewState = newState;
    }
}
```

Comment (jamie94bc) [inline Cirrious.MvvmCross.Droid.Support.AppCompat/MvxActionBarDrawerToggle.cs:9]: Should probably be private setters too?

**Comment by martijn00** [comment]
@jamie94bc can you make a pull request with your version of the MvxActionBarDrawerToggle? It seems you have a better one!

**Comment by jamie94bc** [comment]
@martijn00 Done!

**Comment by martijn00** [comment]
@jamie94bc thanks! I'll close this one.

**Pull request closed without merging**